libraries/aws_efs_file_system.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsEfsFileSystem < AwsResourceBase
+  name 'aws_efs_file_system'
+  desc 'Verifies settings for an EFS File System'
+
+  example "
+    describe aws_efs_file_system('fs-12345678') do
+      it                         { should be_encrypted }
+      its('size_in_bytes.value') { should cmp 6144 }
+    end
+
+    describe aws_efs_file_system(creation_token: 'my-token') do
+      its('encrypted')       { should cmp true }
+      its('throughput_mode') { should eq 'bursting' }
+    end
+  "
+
+  def initialize(opts = {})
+    # Assume the argument is a file_system_id when it is a string type object and reject any other type.
+    opts = { file_system_id: opts } if opts.is_a?(String)
+    opts = {} unless opts.is_a?(Hash)
+
+    # File system identifier can be either the file_system_id or the creation_token.
+    if opts[:file_system_id] && !opts[:file_system_id].empty?
```

Comment:
This could probably be simplified to `opts.key?(:file_system_id)`

Reply:
👍